### ISSUE TITLE
Connector crashes on start device for mlm2pro

### ISSUE THREAD

**Issue by LucMatt** [springbok/MLM2PRO-GSPro-Connector#158]
Connector crashes when trying to start device. I go through all the steps in the https://www.youtube.com/watch?v=R6ix8oRGRh0
video.

Once I hit start to device it briefly searches then the connector crashes or closes out.

**Comment by LucMatt** [comment]
https://www.youtube.com/watch?v=ov2vLTtnCQ4

Actually using this video

**Comment by springbok** [comment]
Are you auto starting GSPro, if so are you using the GSPro launcher app of the gspro.exe app

**Comment by LucMatt** [comment]
No, I'm not starting GSPro. I think the issue is somehow related to the computer in my buildings simulator room. 

I went through the same steps in my apartment with my computer and connected with no issues. Plan on bringing my laptop tomorrow and trying.

**Comment by mdf806** [comment]
I'm having the exact same problem. As soon as I try to start the device it crashes. Any idea what the issue could be?

**Comment by springbok** [comment]
Can you try the new release https://github.com/springbok/MLM2PRO-GSPro-Connector/releases/tag/V1.04.20 and see if you still have this issue.

**Comment by mdf806** [comment]
Will do.  I'll give that one a try this evening.  I'm currently running version V1.04.09 as that one is the most recent one that runs on my pc without crashing.  I'll post my results on the new version tonight.  Thanks for the help!

**Comment by mdf806** [comment]
same issue as previous version.  Crashes as soon as I try to connect the launch monitor.  I'm running windows 10 not 11.  Could that be the issue?

![Image](https://github.com/user-attachments/assets/ed8ed9bc-c624-4896-8bf9-249f5166a99f)

**Comment by springbok** [comment]
Are you able to upload the log file in appdata\logs directory after it stops working.It should run on windows 10 or 11 without any problems.Sent from my iPhoneOn 18 Jan 2025, at 11:11 am, mdf806 ***@***.***> wrote:﻿
same issue as previous version.  Crashes as soon as I try to connect the launch monitor.  I'm running windows 10 not 11.  Could that be the issue?
image.png (view on web)

**Comment by mdf806** [comment]
[mlm2pro-gspro-connect.log](https://github.com/user-attachments/files/18465171/mlm2pro-gspro-connect.log)

log file attached

**Comment by als352** [comment]
Wondering if there was a resolution to the issue noted above? I have the same issue, though I am admittedly trying to run the connector within Crossover in MacOS. I do not own a gaming pc (or any Windows PC), so am hoping to get this to work to use GSPro with my MLM2Pro. 

Within Crossover, GSPro launches without issue, I can open the connector, and can connect to GSPro OpenAPI. But as soon as I click connect to Rapsodo BT the connector application crashes.

**Comment by springbok** [comment]
> Wondering if there was a resolution to the issue noted above? I have the same issue, though I am admittedly trying to run the connector within Crossover in MacOS. I do not own a gaming pc (or any Windows PC), so am hoping to get this to work to use GSPro with my MLM2Pro.
> 
> Within Crossover, GSPro launches without issue, I can open the connector, and can connect to GSPro OpenAPI. But as soon as I click connect to Rapsodo BT the connector application crashes.

Here are some instruction from Peter Blair a user that Crossover to work:

There are four main steps for running GSPro and the MLM2Pro on your Apple-series laptop:
1.	Install Crossover and install GSPro into a bottle in Crossover
2.	Install UTM and Windows 11 into UTM - make sure to enable shared network settings (https://www.youtube.com/watch?v=hsrOrqplNBo)
3.	Install Springbok into UTM and configure it
4.	Turning everything on in the right order and connecting
•	Go into your router settings, view the list of attached devices and see what IP address is assigned to your Mac (typically 10.0.0.x with x being a number)  
•	Set up static IP for your Mac (i.e. so that it will always have the same IP address to be shared from UTM to Crossover 10.0.0.3)
•	Convert your GSPro connector to OpenAPI here https://gsprogolf.com/convert.html
•	Turn on your MLM2Pro and open the Rapsodo MLM2Pro app on your phone/tablet
•	Once connected, go to 3rd Party Apps and click Awesome Golf. Click authenticate (you will need to do this every 24 hours).
•	Go to Practice Range or any other Rapsodo Play Now mode to align your launch monitor
•	Disconnect the MLM2Pro from the app (light should turn red)
•	Turn Mac Bluetooth off - DO NOT SKIP THIS STEP
•	Start Crossover and open GSPro - make sure you click "reset GSPro Connection" the first time you do this
•	Open UTM and open your Windows 11 virtual machine that has the Springbok connector installed
•	Plug in your USB bluetooth dongle and say yes to the two dialogue boxes about allowing it to be viewed/used by the virtual machine in UTM.
•	Open Springbok in UTM. Click the settings gear cog.
•	In the far left side under GSPro, enter the fixed local IP address you entered earlier.
•	Make sure Launch Monitor is set to Rapsodo MLM2Pro BT.
•	Click Connect to GSPro in Springbok - it should almost immediately turn green. If it freezes this means you have the IP address for your Mac incorrect.
•	Click Connect to MLM2Pro in Springbok - it should go through a series of steps and the MLM2Pro light should eventually turn green.
•	Go into GSPro and start a course/session.

NB - The type of Bluetooth dongle you use seems to matter. The Panda Bluetooth 4.0 USB Nano Adapter worked for me, after finding it on a listing of approved Bluetooth dongles for RaspberryPi integration.

You may also need to play with resolution and graphic quality settings to get it running smoothly on your Mac.
